### PR TITLE
interfaces: add DisconnectSnap

### PR DESCRIPTION
**Pull request by zyga**
This patch adds a way to disconnect all connections to / from a given
snap. This is useful for performing snap upgrades where all connections
are severed before the snap is replaced.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: interfaces/repo.go [hunk 1]
```diff
@@ -656,3 +656,35 @@ func (r *Repository) RemoveSnap(snapInfo *snap.Info) error {
 
 	return nil
 }
+
+// DisconnectSnap disconnects all the connections to and from a given snap.
+//
+// The return value is a list of snap.Info's that were affected.
```
Comment (niemeyer) [inline interfaces/repo.go]: Why do we need to return this?  We can only have one possible connected snap, the one that is currently active with the given name.

Comment (zyga) [inline interfaces/repo.go]: We want to know this because then we know that those snaps should have their security configured.

Comment (niemeyer) [inline interfaces/repo.go]: The only reason to call this function is because we want the known (!) snap reconfigured.

Comment (zyga) [inline interfaces/repo.go]: Perhaps we disagree on how this is intended to be used:

The ``setup-security`` task will:
 - call repo.DisconnectSnap()
 - call repo.RemoveSnap()
 - call repo.AddSnap()
 - rebuild connections based on what is in the state (and update state if neccessary)
 - call backend.Setup() for each backend x affected snap

We cannot reconfigure just one snap. Even if we break this down into separate tasks, we have to know the affected set to properly react to snap changes.

Comment (niemeyer) [inline interfaces/repo.go]: You of course totally right, sorry. The problem isn't the local snap being removed, but the other ones (stating to myself). Thanks.

**Comment by niemeyer** [comment]
LGTM

**Pull request merged**